lldb/source/Interpreter/CommandInterpreter.cpp

```diff
@@ -1883,9 +1886,49 @@ bool CommandInterpreter::HandleCommand(const char *command_line,
                                        LazyBool lazy_add_to_history,
                                        CommandReturnObject &result,
                                        bool force_repeat_command) {
+  lldb_private::telemetry::ScopedDispatcher<
+      lldb_private::telemetry::CommandInfo>
+      helper(&m_debugger);
+  lldb_private::telemetry::TelemetryManager *ins =
+      lldb_private::telemetry::TelemetryManager::GetInstance();
+  const int command_id = ins->MakeNextCommandId();
+
   std::string command_string(command_line);
   std::string original_command_string(command_string);
   std::string real_original_command_string(command_string);
+  std::string parsed_command_args;
+  CommandObject *cmd_obj = nullptr;
```

Comment:
I really dislike that the RAII object forces us to declare variables early like it's C99. Do we need the RAII helper here?  Are there early returns we need to account for, or can we just fill in the object at the end of the function?

Reply:
Yes, unfortunately, there are a bunch of early returns in this function (which is also very long).
- Line 1941-1945 : return due to some interruption
- Line 1997 - 2020 : nested if  statements to set the right return-status.